Tests/GitHubReleases.tests.ps1

```diff
@@ -12,78 +12,148 @@ $moduleRootPath = Split-Path -Path $PSScriptRoot -Parent
-                $tagged = @(Get-GitHubRelease -OwnerName $ownerName -RepositoryName $repositoryName -Tag $releases[$taggedIndex].tag_name)
+        Context 'When getting a tagged release' {
+            $taggedIndex = 8
+            $tagged = @(Get-GitHubRelease -OwnerName $ownerName -RepositoryName $repositoryName -Tag $releases[$taggedIndex].tag_name)
 
-                It 'Should return one value' {
-                    $tagged.Count | Should Be 1
-                }
+            It 'Should return one value' {
+                $tagged.Count | Should Be 1
+            }
 
-                It 'Should return the correct release' {
-                    $tagged.name | Should Be $releases[$taggedIndex].name
-                }
+            It 'Should return the correct release' {
+                $tagged.name | Should Be $releases[$taggedIndex].name
             }
         }
+    }
 
-        Describe 'Getting releases from default owner/repository' {
-            $originalOwnerName = Get-GitHubConfiguration -Name DefaultOwnerName
-            $originalRepositoryName = Get-GitHubConfiguration -Name DefaultRepositoryName
+    Describe 'Getting releases from default owner/repository' {
+        $originalOwnerName = Get-GitHubConfiguration -Name DefaultOwnerName
+        $originalRepositoryName = Get-GitHubConfiguration -Name DefaultRepositoryName
 
-            try {
-                Set-GitHubConfiguration -DefaultOwnerName "dotnet"
-                Set-GitHubConfiguration -DefaultRepositoryName "core"
-                $releases = @(Get-GitHubRelease)
+        try
+        {
+            Set-GitHubConfiguration -DefaultOwnerName "dotnet"
+            Set-GitHubConfiguration -DefaultRepositoryName "core"
+            $releases = @(Get-GitHubRelease)
 
-                Context 'When getting all releases' {
-                    It 'Should return multiple releases' {
-                        $releases.Count | Should BeGreaterThan 1
-                    }
+            Context 'When getting all releases' {
+                It 'Should return multiple releases' {
+                    $releases.Count | Should BeGreaterThan 1
                 }
-            } finally {
-                Set-GitHubConfiguration -DefaultOwnerName $originalOwnerName
-                Set-GitHubConfiguration -DefaultRepositoryName $originalRepositoryName
             }
         }
+        finally
+        {
+            Set-GitHubConfiguration -DefaultOwnerName $originalOwnerName
+            Set-GitHubConfiguration -DefaultRepositoryName $originalRepositoryName
+        }
+    }
+
+    Describe 'Creating, changing and deleting releases' {
+        $repo = New-GitHubRepository -RepositoryName ([Guid]::NewGuid().Guid) -AutoInit -Private
+
+        Context 'When creating a simple new release' {
+            $release = New-GitHubRelease -Uri $repo.svn_url -TagName $script:defaultTagName
+            $queried = Get-GitHubRelease -Uri $repo.svn_url -Release $release.id
```

Comment:
Pester 5 support is being tracked by #194.  Definitely desiring to have that support added.  I haven't yet looked into what's involved for that though.

I agree with you that this setup work belongs in `BeforeAll`/`AfterAll`.  At the time, I forgot about those and specifically wanted to avoid `BeforeEach`/`AfterEach` due to the added time from unnecessary duplicated setup that they would incur.  Will look into moving setup/teardown as appropriate.

Thanks.